integrations/terraform/hashicorp-vault/README.md

```diff
@@ -0,0 +1,219 @@
+
+## Retrieving a Secret from Vault
+
+Now you can use the `vault_generic_secret` data source to retrieve the secret from Vault:
+
+```hcl
+data "vault_generic_secret" "materialize_password" {
+  path = "secret/materialize"
+}
+```
+
+Finally, you can use the secret retrieved from Vault in your `materialize_secret` resource:
+
+```hcl
+resource "materialize_secret" "example_secret" {
+  name  = "pgpass"
+  value = data.vault_generic_secret.materialize_password.data["pgpass"]
+}
+```
+
+## Running the Terraform Script
+
+Now that you have the Terraform script ready, you can run the following commands to initialize and apply the script:
+
+```bash
+# Initialize the Terraform configuration:
+terraform init
+
+# Apply the Terraform configuration:
+terraform apply
+```
+
+Output:
+
+```bash
+Plan: 1 to add, 0 to change, 0 to destroy.
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value: yes
+
+materialize_secret.example_secret: Creating...
+materialize_secret.example_secret: Creation complete after 3s [id=u111]
+
+Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
+```
+
+Once the script has been applied, you can verify that the secret has been created in Materialize Cloud:
+
+```sql
+SHOW SECRETS;
+```
+
+## Using the Secret in Connections
+
+Once the secret has been created, you can use it in your `CREATE CONNECTION` statements. In this example, we'll create a connection to a Postgres instance using the secret we created using the `materialize_connection_postgres` resource:
+
+```hcl
+# Create a Postgres Connection
+resource "materialize_connection_postgres" "example_postgres_connection" {
+  name = "example_postgres_connection"
+  host = "instance.foo000.us-west-1.rds.amazonaws.com"
+  port = 5432
+  user {
+    text = "pguser"
+  }
+  password {
+    name          = materialize_secret.example_secret.name
+    database_name = "materialize"
+    schema_name   = "public"
+  }
+  database = "pgdatabase"
+}
+```
+
+The `password` block references the secret we created earlier.
+
+## Cleaning up
+
+To clean up the resources created by Terraform, you can run the following command:
+
+```bash
+terraform destroy
+```
+
+## Conclusion
+
+In this demo, you learned how to use the Materialize Terraform provider to create a secret in Materialize Cloud using a secret stored in HashiCorp Vault.
+
+In a production environment, you wouldn't run Vault locally in dev mode. Instead, a highly available, secure, and resilient Vault cluster would be deployed on your cloud instances, ensuring that secrets remain accessible and secure. Rather than using the root token for authentication, you'd employ more secure authentication methods, such as IAM roles for AWS or Kubernetes service accounts and all communication between Vault and its clients should be over a secure HTTPS connection.
```

Comment:
```suggestion
This demo used Vault in development mode, which **should never be used in production environments**. Instead, follow the [HashiCorp documentation](https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-deploy) to deploy a highly available, secure, and resilient Vault cluster on your cloud instances, ensuring that secrets remain accessible and secure. Rather than using the root token for authentication, you'd employ more secure authentication methods, such as IAM roles for AWS or Kubernetes service accounts, and all communication between Vault and its clients should be over a secure HTTPS connection.
```

Reply:
We should add the "never, ever use dev mode in production" disclaimer to the step where it is used, just to be sure!